apps/api/src/app/events/usecases/add-job/add-job.usecase.ts

```diff
@@ -57,10 +71,16 @@ export class AddJob {
       })
     );
 
+    if (delay === null) {
+      Logger.warn('variable delay is null which is not apart of the definition');
+    }
+
+    Logger.verbose('Adding Job to Queue');
     await this.workflowQueueService.addToQueue(job._id, job, delay);
 
     if (delay) {
-      this.createExecutionDetails.execute(
+      Logger.verbose('Delay is active, Creating execution details');
+      await this.createExecutionDetails.execute(
```

Comment:
@Cliftonz those are not awaited by design here to avoid blocking the response to the user